app/src/studio/behavior_subsystem.c

```diff
@@ -0,0 +1,216 @@
+            desc.which_value_type = zmk_behaviors_BehaviorParameterValueDescription_hid_usage_tag;
+#if IS_ENABLED(CONFIG_ZMK_HID_CONSUMER_REPORT_USAGES_BASIC)
+            desc.value_type.hid_usage.consumer_max = 0xFF;
+#else
+            desc.value_type.hid_usage.consumer_max = 0xFFF;
+#endif
+
+#if IS_ENABLED(CONFIG_ZMK_HID_REPORT_TYPE_NKRO)
+            desc.value_type.hid_usage.keyboard_max = CONFIG_ZMK_HID_REPORT_TYPE_NKRO;
+#else
+            desc.value_type.hid_usage.keyboard_max = 0xFF;
+#endif
+            break;
+        case BEHAVIOR_PARAMETER_VALUE_TYPE_LAYER_ID:
+            desc.which_value_type = zmk_behaviors_BehaviorParameterValueDescription_layer_id_tag;
+            break;
+        default:
+            LOG_ERR("Unknown value description type %d", val->type);
+            return false;
+        }
+
+        if (!pb_encode_submessage(stream, &zmk_behaviors_BehaviorParameterValueDescription_msg,
+                                  &desc)) {
+            LOG_WRN("Failed to encode submessage!");
+            return false;
+        }
+    }
+
+    return true;
+}
+
+static bool encode_metadata_sets(pb_ostream_t *stream, const pb_field_t *field, void *const *arg) {
+    struct encode_metadata_sets_state *state = (struct encode_metadata_sets_state *)*arg;
+    bool ret = true;
+
+    LOG_DBG("Encoding the %d metadata sets with %p", state->sets_len, state->sets);
+
+    for (int i = 0; i < state->sets_len; i++) {
+        LOG_DBG("Encoding set %d", i);
+        if (!pb_encode_tag_for_field(stream, field))
+            return false;
+
+        state->i = i;
+        zmk_behaviors_BehaviorBindingParametersSet msg =
+            zmk_behaviors_BehaviorBindingParametersSet_init_zero;
+        msg.param1.funcs.encode = encode_value_description;
+        msg.param1.arg = state;
+        msg.param2.funcs.encode = encode_value_description;
+        msg.param2.arg = state;
+        ret = pb_encode_submessage(stream, &zmk_behaviors_BehaviorBindingParametersSet_msg, &msg);
+        if (!ret) {
+            LOG_WRN("Failed to encode submessage for set %d", i);
+            break;
+        }
+    }
+
+    return ret;
+}
+
+static bool encode_behavior_name(pb_ostream_t *stream, const pb_field_t *field, void *const *arg) {
+    struct zmk_behavior_ref *zbm = (struct zmk_behavior_ref *)*arg;
+
+    if (!pb_encode_tag_for_field(stream, field))
+        return false;
+
+    return pb_encode_string(stream, zbm->metadata.display_name, strlen(zbm->metadata.display_name));
+}
+
+static struct encode_metadata_sets_state state = {};
+
+zmk_Response get_behavior_details(const zmk_Request *req) {
+    uint32_t behavior_id = req->subsystem.behaviors.request_type.get_behavior_details.behavior_id;
+
+    const char *behavior_name = zmk_behavior_find_behavior_name_from_local_id(behavior_id);
+
+    if (!behavior_name) {
+        LOG_WRN("No behavior found for ID %d", behavior_id);
+        return ZMK_RPC_SIMPLE_ERR(GENERIC);
```

Comment:
What does the `type` parameter to this macro do? I don't see any definition of `GENERIC` in this PR.